New Entry: C++ Math Functions: `sqrt()` (#974)
Codecademy/docs: content/cpp/concepts/math-functions/terms/sqrt/sqrt.md

---
Title: 'sqrt()'

> Description: 'Returns square root of the argument'

Reviewer: ```suggestion
Description: 'Returns square root of the argument.'
```

Subjects:
  - 'Computer Science'
Tags:
  - 'Functions'
  - 'Arithmetic'
CatalogContent:
  - 'learn-c-plus-plus'
  - 'paths/computer-science'
---

> Returns square root of the argument

Reviewer: ```suggestion
The **`sqrt()`** function returns the square root of the argument.
```

Contrary to the current state/formatting of existing terms under [C++: Math Functions](https://www.codecademy.com/resources/docs/cpp/math-functions), the suggestions here aim for a new formatting, including:

- Beginning the entry by referring to the term by name.
- **Bolding** the first mention of the concept or term.

## Syntax

```cpp
 sqrt(x)
```

> 

Reviewer: ```suggestion

If the parameter `x` is negative (less than 0), then a domain error will occur.

```

## Example

use `sqrt()` to find the square root of `4`:

```cpp
#include <iostream>
#include <cmath>

int main() {
  double x = 4;
  double result;

  result = std::sqrt(x);

  std::cout << "The square root of " << x << " is " << result << "\n";

>   // Output: The square root of 4 is 2

Reviewer: ```suggestion
  // Output: The square root of 4 is 2
  }
```

Syntax error occurs without this closing curly brace.

```

## Codebyte Example

```codebyte/cpp
#include <iostream>
#include <cmath>

int main() {
  double x = 9;
  double result;

  result = std::sqrt(x);

>   std::cout << "The square root of " << x << " is " << result << "\n";
>   // Output: The square root of 9 is 3

Reviewer: ```suggestion
  std::cout << "The square root of " << x << " is " << result << "\n";
```

Since this is part of a runnable Codebyte example, the output will be rendered so there is no need for the comment.

---
Title: 'sqrt()'
Description: 'Returns square root of the argument.'
Subjects:
  - 'Computer Science'
Tags:
  - 'Functions'
  - 'Arithmetic'
CatalogContent:
  - 'learn-c-plus-plus'
  - 'paths/computer-science'
---

The **`sqrt()`** function returns the square root of the argument.

## Syntax

```cpp
sqrt(x)
```

If the parameter `x` is negative (less than 0), then a domain error will occur.

## Example

> use `sqrt()` to find the square root of `4`:

Reviewer: ```suggestion
The following example uses the `sqrt()` function to find the square root of `4`:
```

Reviewer: The old text is more instructional in tone; Docs entries should be more informational/declarative.

```cpp
#include <iostream>
#include <cmath>

int main() {
  double x = 4;
  double result;

  result = std::sqrt(x);

  std::cout << "The square root of " << x << " is " << result << "\n";
  // Output: The square root of 4 is 2

>   }

Reviewer: ```suggestion
}
```

```

## Codebyte Example

> 

Reviewer: ```suggestion

The following example is runnable and returns the square root of `9` with the `sqrt()` function:

```

Reviewer: It's preferable to preface any code snippet with 1-2 sentences explaining what's going on.

---
Title: 'sqrt()'
Description: 'Returns square root of the argument.'
Subjects:
  - 'Computer Science'
Tags:
  - 'Functions'
  - 'Arithmetic'
CatalogContent:
  - 'learn-c-plus-plus'
  - 'paths/computer-science'
---

The **`sqrt()`** function returns the square root of the argument.

## Syntax

```cpp
sqrt(x)
```

If the parameter `x` is negative (less than 0), then a domain error will occur.

## Example

The following example uses the `sqrt()` function to find the square root of `4`:

```cpp
#include <iostream>
#include <cmath>

int main() {
  double x = 4;
  double result;

  result = std::sqrt(x);

  std::cout << "The square root of " << x << " is " << result << "\n";
  // Output: The square root of 4 is 2
}
```

## Codebyte Example

> The following example is runnable and returns the square root of `9` with the `sqrt()` function:

Reviewer: ```suggestion
The following example is runnable and returns the square root of `9` with the `sqrt()` function:

```

Author: It's saying the suggestion is identical

Reviewer: Ah, that it is!

Reviewer: Actually, this suggestion still stands. In a [previous suggestion](https://github.com/Codecademy/docs/pull/974/files#r966240255), it recommend empty above and below line # 45. However, there is still no empty line between # 45 and # 46.

Reviewer: If you still run into issues, I'm happy to move in and make this change 😄

Author: It won't let me just commit your suggestion, saying it's identical

Reviewer: No worries! Made the change in 350a39158608949bb323f0007f280a6de6359e76. All the checks should pass now 😄
